tests/src/python/test_qgsserver.py

```diff
@@ -330,8 +330,8 @@ def test_requestHandler(self):
         request = QgsBufferServerRequest('http://somesite.com/somepath', QgsServerRequest.GetMethod, headers)
         response = QgsBufferServerResponse()
         self.server.handleRequest(request, response)
-        self.assertEqual(bytes(response.body()), b'<ServerException>Project file error</ServerException>\n')
-        self.assertEqual(response.headers(), {'Content-Length': '54', 'Content-Type': 'text/xml; charset=utf-8'})
+        self.assertEqual(bytes(response.body()), b'<ServerException>Project file error. For OWS services: please provide a SERVICE and a MAP parameter pointing to a valid QGIS project file</ServerException>\n')
```

Comment:
Actually I'm not fully convinced about this. I mean, I can live with that, but exposing `MAP` in the query string URL is not only bad practice but it is also violating WFS3 specs (I know this code is not going to be hit by WFS3, but still).

So, if you really wanted to be precise you should list the other preferred methods that are available to make QGIS aware of the project it needs to serve, but that is probably too much for an error message.

Reply:
I agree, it's just that: telling people that there is a "Project file error" if they are trying to get the server working seems not appropriate. If they add a MAP and SERVICE param, it will give them futher hints...

Reply:
It depends who "people" are. 
Who looks at the XML output of a machine 2 machine designed service must be a special kind of people that should also be prepared to look at the server logs and RTFM.

The server is **not** an end user application (not yet).

IMHO a better message would be: "A valid QGIS project could not be loaded. Please check server configuration." 

Because even "file" is wrong wording: what if the server is configured to use projects stored indide a DB?

Reply:
Hi Gents, never mind, I think we just differ in what a server should and should not tell the user. 

QGIS in my opinion hides everything for the average user. Other WMS's like Minnesota Mapserver and Geoserver tell the user if they have configured something wrong.

I think OGC servers are not only there to be used for machine<->machine communication. It all starts with average users who try to setup servers and fail.

I've tried and failed :-)

Reply:
@rduivenvoorde believe me when I say that I understand your frustration, I get my daily share of it.

In this PR there is value (I would have already merged the logging order change for instance).

But if the current message "Project file error" is wrong, replacing it with another slightly better but still wrong message is not what we want.

Would it be ok for you to change it to something that is more descriptive but also correct?

What comes to my mind is:
- A valid QGIS project could not be loaded. This may be the result of an invalid project or a missing project identifier, please refer to the QGIS Server logs and the official server documentation for further details.

Also note that when the project is valid and the `SERVICE` is missing you get a different error: 
```xml
<ServiceException code="Service configuration error">Service unknown or unsupported</ServiceException>
```
Which is a 200OK response and not (like in the previous case) an internal server error with a 500 code.

And I insist: if the average user (whoever she is) is trying to set up a web server she must be prepared to look at the server logs (where there are much more details about the error).

Reply:
@elpaso it all started with this change of order, which was easy. All other stuff took me considerable (learning) time, for example to find out how to run the tests (still not able to run them in Qt-Creator). You are probably right, but it takes me too much time already. Maybe just cherry-pick the first commit and leave rest as is?

Reply:
thanks @rduivenvoorde 
I'll do the update

Reply:
> @elpaso it all started with this change of order, which was easy. All other stuff took me considerable (learning) time, for example to find out how to run the tests (still not able to run them in Qt-Creator). 

That might be tricky, I can attach my config as an example:

![immagine](https://user-images.githubusercontent.com/142164/80735494-0ef71480-8b11-11ea-84be-ea509ede9217.png)

And my env:
```
LD_LIBRARY_PATH=/home/xxx/local5_13_2_release_gdal3/lib/:/home/xxx/dev/oracle/instantclient_19_3
OGR_GPKG_FOREIGN_KEY_CHECK=NO
PYTHONPATH=/home/xxx/local5_13_2_release_gdal3/lib/python3/dist-packages:/home/xxx/dev/QGIS/build-local-qt5132-release_gdal3/output/python/:/home/xxx/dev/QGIS/build-local-qt5132-release_gdal3/output/python/plugins:/home/xxx/dev/QGIS/tests/src/python/:/home/xxx/local13_2_release_gdal3/lib/python3/dist-packages/:/home/xxx/.local/lib/python3.6/site-packages/:/usr/lib/python3.6:/usr/lib/python3.6/plat-x86_64-linux-gnu:/usr/lib/python3.6/lib-dynload:/home/xxx/.local/lib/python3.6/site-packages:/usr/local/lib/python3.6/dist-packages:
QGIS_PREFIX_PATH=/home/xxx/dev/QGIS/build-local-qt5132-release_gdal3/output
```

> You are probably right, but it takes me too much time already. Maybe just cherry-pick the first commit and leave rest as is?

Whatever works for you, I'll approve (actually I had already approved this one, I just hoped for a better error message).

Reply:
@elpaso thanks. Actually I do not know who is doing the merge after an approval, I can do it myself, as I have commit rights, but I thought that a select group of core committers actually did all the merges... Was I supposed to do it myself?

Reply:
Once it's approved you can merge it by yourself, but in this case this PR has been already merged by @rldhont 

Thank you for your patience :)

Reply:
@rduivenvoorde thanks for your work, I have merge it because it has been approved by @elpaso.